frontend/benefit/applicant/src/components/applications/forms/application/step4/useApplicationFormStep4.ts

```diff
@@ -0,0 +1,88 @@
+import hdsToast from 'benefit/applicant/components/toast/Toast';
+import ApplicationContext from 'benefit/applicant/context/ApplicationContext';
+import useUpdateApplicationQuery from 'benefit/applicant/hooks/useUpdateApplicationQuery';
+import { useTranslation } from 'benefit/applicant/i18n';
+import {
+  Application,
+  ApplicationData,
+} from 'benefit/applicant/types/application';
+import { getApplicationStepString } from 'benefit/applicant/utils/common';
+import { TFunction } from 'next-i18next';
+import React, { useEffect, useState } from 'react';
+import snakecaseKeys from 'snakecase-keys';
+
+type ExtendedComponentProps = {
+  t: TFunction;
+  handleNext: () => void;
+  handleBack: () => void;
+  handleStepChange: (step: number) => void;
+  translationsBase: string;
+};
+
+const useApplicationFormStep4 = (
```

Comment:
Actually shared `ApplicationWizard` component is now in develop, so if you start to use it, then inside steps you can use `useWizard` -hook which tackle some handle some common logic like handling step changes.

Reply:
i have a feeling that it wont be that easy to integrate wizard, to much refactoring would be needed and we have a lot of corner cases in validation

Reply:
the data only is the same, but presentation is completely different. even the attachment list has 2 implementations for view and edit, so we dont have too much conditional logic there.

Reply:
> i have a feeling that it wont be that easy to integrate wizard, to much refactoring would be needed and we have a lot of corner cases in validation

I think that the validation should not be part of the wizard logic, you can give any async or non-async function to `handleStep` function in `const {handleStep} = useWizard()` as a parameter and it wont proceed to next step if function throws validation or some other error. You can use also `Wizard` component instead of `ApplicationWizard` which has only the core functionality.

Reply:
we will give it a try definetly